### PR TITLE
Account for TZ in EvidenceSubmissionWindow End Date during Substitution

### PR DESCRIPTION
**Pull request by jcq**
Connects [CASEFLOW-1568](https://vajira.max.gov/browse/CASEFLOW-1568)

### Description
This updates the logic for calculating the end date of a new `EvidenceSubmissionWindowTask` created during appellant substitution to be set at midnight of the user's time zone, not UTC. Previously it was saving as midnight UTC, so to users in the US, it appeared as being one day earlier than anticipated.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] New `EvidenceSubmissionWindowTask.timer_ends_at` is midnight w/ user's TZ offset

### Testing Plan
1. Log in as `COB_USER`
2. Perform appellant substitution for Appeal with `evidence_submission` docket type
3. Be sure to select the `EvidenceSubmissionWindowTask` during substitution
4. Ensure that `timer_ends_at` on the new `EvidenceSubmissionWindowTask` returns a value with appropriate TZ offset (not UTC)

### CODE REVIEW

File: spec/feature/queue/substitute_appellant/shared_setup.rb [hunk 1]
```diff
@@ -152,6 +152,17 @@
...
 
+      # Verify that the Evidence Submission Window was stored correctly
+      if docket_type.eql?("evidence_submission")
+        # Ensure that our new window ends on specified date, accounting for user's time zone (not based on midnight UTC)
+        window_task = EvidenceSubmissionWindowTask.find_by(appeal: new_appeal)
+        expect(window_task.timer_ends_at).to be_between(
+          evidence_submission_window_end_time - 1.day,
+          evidence_submission_window_end_time + 1.day
+        )
+        expect(window_task.timer_ends_at.utc_offset).to eql(Time.zone.now.utc_offset)
```
Comment (eileen-nava) [inline spec/feature/queue/substitute_appellant/shared_setup.rb:163]: The testing is strong with this PR.

**Review by eileen-nava — APPROVED** [review]
I'm approving this.  I reviewed it with breakpoints to dig into the code a bit, and the testing is thorough.

That being said, when I followed the testing plan and made an appeal with a docket type of evidence submission, the appeal did not have an evidence submission window task.  JC, did you separately create an evidence submission window task for the appeal?  I'm curious because when I did a [previous PR for this work](https://github.com/department-of-veterans-affairs/caseflow/pull/16362), I couldn't get the evidence submission window task factory to work unless I commented out code.

TLDR: I did not do a manual review, however, I am confident in the code.

**Comment by eileen-nava** [comment]
@jcq Please update the [relevant part of the wiki](https://github.com/department-of-veterans-affairs/caseflow/wiki/Frontend-Things-to-Avoid#new-date) with instructions about how to manage time zone offsets.  Thanks!
